src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs

```diff
@@ -120,8 +120,13 @@ protected virtual string GeneratePathFromRoute(string routeName, IDictionary<str
         }
 
         /// <inheritdoc />
-        public virtual string Content([NotNull] string contentPath)
+        public virtual string Content(string contentPath)
         {
+            if (contentPath == null)
+            {
+                return null;
+            }
+
             return GenerateClientUrl(_httpContext.Request.PathBase, contentPath);
```

Comment:
Not sure if we need to fix this here, but one of the reasons for #2841 is that `GenerateClientUrl` isn't trimming the `contentPath` so a leading space will result in `~` not being removed.
Do we want to fix that here or leaving the trimming up to caller?